tableau_utilities/scripts/connection.py

```diff
@@ -126,14 +111,17 @@ def embed_credential_online(datasource_id, conn_settings, server, debugging_logs
 
 
 def connection(args, server=None):
-    """
+    """ Updates a Datasource's connection in the file locally
 
+    Args:
+        args: An argparse args object
+        server (TableauServer): The TableauServer object
     """
 
     # Set Args
-    datasource_id = args.id
-    name = args.name
-    project_name = args.project_name
+    datasource_id = args.id if args.id else server.get_datasource(
```

Comment:
@JustinGrilli. I don't think we want to get the datasource id here. not all operations need it so it makes an API call that's unnecessary part of the time up here

Reply:
It will only make the API call if it reaches the `else`, which would only happen if the ID was not provided.

Reply:
I made this statement a little more intuitive to read! 🙇